pkg/webhook/shoot/loadbalancer_service.go

```diff
@@ -8,11 +8,20 @@ import (
 	"context"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (m *mutator) mutateService(_ context.Context, service *corev1.Service) error {
+func (m *mutator) mutateService(ctx context.Context, service *corev1.Service, shootClient client.Client) error {
 	if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
-		for _, v := range service.Spec.IPFamilies {
+		if service.Annotations == nil {
+			service.Annotations = make(map[string]string, 1)
+		}
+		kubeDNSService := &corev1.Service{}
+		if err := shootClient.Get(ctx, types.NamespacedName{Name: "kube-dns", Namespace: "kube-system"}, kubeDNSService); err != nil {
+			return err
+		}
+		for _, v := range kubeDNSService.Spec.IPFamilies {
```

Comment:
Should we also add a check for internal load balancers? With the current approach we make all load balancers in a dual stack cluster automatically externally visible.